our-paxos/paxos.py

```diff
@@ -193,43 +198,110 @@ def proposer(config, id):
-            while(id - 1 > len(messages)):
-                messages.append(-1)
-            messages.append(value)
-        # If id is = len(msg) we apppend a message to the array,
-        # we don't show that we learned anything if learned isn't equal to msg len
-        # because in this case we're still missing message somewhere in the middle
-        elif id == len(messages):
-            if learned == len(messages):
-                print("Learned: ", value)
-                learned+=1
-            messages.append(value)
-
-        # If id is < len(msg) we swap -1 (value of not received message) with the received value,
-        # then if id == learned, then we can print the value as learned until we reach first
-        # undefined message
-        elif id < len(messages):
-            messages[id] = value
-            if id == learned:
-                while(messages[learned] != -1 and len(messages) > learned):
-                    print("Learned: ", value)
-                    learned += 1
-
+        inst_id = msg[0]
+        if len(msg) > 1:
+            # If we receive un "update message" from another learner
+            if msg[1] == 1:
+                while inst_id <= len(messages):
+                    messages.append(-1)
+                if messages[inst_id] == -1:
+                    inst_id[inst_id] = msg[3]
+
+            # If we received PAXOS round 2 message
+            if(msg[1] == 2):
+                # print("Instance: {} Learned: {}".format(inst_id, msg[3]))
+                value = msg[3]
+                # If id is > len(msg) we need to extend array of messages up to the needed size,
+                # we don't show that we learned anything, bc we've extended array without assigning
+                # the message with the smaller id
+                if inst_id > len(messages):
+                    print("Init mor: {}".format(inst_id))
+                    while(inst_id - 1 > len(messages)):
+                        messages.append([])
+                        messages_running.append(False)
+                    messages.append([value])
+                    messages_running.append(True)
+                # If id is == len(msg) we apppend a message to the array,
+                # we don't show that we learned anything if learned isn't equal to msg len
+                # because in this case we're still missing message somewhere in the middle
+                elif inst_id == len(messages):
+                    print("Init eq: {}".format(inst_id))
+                    # if learned == len(messages) and len(messages[id]) == QUORUM_AMOUNT - 1:
+                    #     if value == messages[id][0]:
+                    #         print("Instance: {} Learned: {}".format(id, value))
+                    #     learned += 1
+                    messages.append([value])
+                    messages_running.append(True)
+
+                # If id is < len(msg) we swap -1 (value of not received message) with the received value,
```

Comment:
Just to clarify, the value of a "not received message" is [] now and not "-1", right?

Reply:
yup, fixed